components/requests/requests.go

```diff
@@ -25,15 +26,21 @@ func Init(hostUrl *url.URL, tokenValue string) {
 }
 
 func Get(path string, query *Query) (responseBody []byte, err error) {
-	return Do("GET", path, query, nil)
+	loadingFunc := func() ([]byte, error) { return Do("GET", path, query, nil) }
+
+	return printer.WithSpinner(loadingFunc)
 }
 
 func Post(path string, requestData *RequestData) (responseBody []byte, err error) {
-	return Do("POST", path, nil, requestData)
+	loadingFunc := func() ([]byte, error) { return Do("POST", path, nil, requestData) }
+
+	return printer.WithSpinner(loadingFunc)
 }
 
 func Patch(path string, requestBody *RequestData) (responseBody []byte, err error) {
-	return Do("PATCH", path, nil, requestBody)
+	loadingFunc := func() ([]byte, error) { return Do("PATCH", path, nil, requestBody) }
+
+	return printer.WithSpinner(loadingFunc)
 }
 
 func Do(method string, path string, query *Query, requestData *RequestData) (responseBody []byte, err error) {
```

Comment:
I think, we should reduce visibility of the `Do` function to `do`, then we can assure, that every requests uses the spinner.

the 100ms used in `loading.go`, is that the debounce?

Reply:
By go itself: https://stackoverflow.com/questions/24790175/when-is-the-init-function-run